infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -73,6 +100,140 @@ public boolean isPredefined() {
     return isPredefined;
   }
 
+  /**
+   * Creates a Kubernetes namespace for the specified workspace.
+   *
+   * <p>Namespace won't be prepared. This method should be used only in case workspace recovering.
+   *
+   * @param workspaceId identifier of the workspace
+   * @return created namespace
+   */
+  public KubernetesNamespace create(String workspaceId, String namespace) {
+    return doCreateNamespace(workspaceId, namespace);
+  }
+
+  @VisibleForTesting
+  KubernetesNamespace doCreateNamespace(String workspaceId, String name) {
```

Comment:
What is the purpose of `doCreateNamespace`? It looks like its parameters are transparently passed from `create`.

Reply:
```java
@VisibleForTesting
KubernetesNamespace doCreateNamespace(String workspaceId, String name)
```

It's designed in such a way to be able to inject `KubernetesNamespace` mock